Templates/Android/generated/BaseMethodRequestBuilder.java.tt

```diff
@@ -5,26 +5,33 @@
 <#host.TemplateName = BaseTypeRequestBuilder(c);#>
 <#=writer.WriteHeader()#>
 <#=CreatePackageDef(host)#>
+<#
+var methods = new List<OdcmMethod>();
+methods.Add(c.AsOdcmMethod());
+methods.AddRange(c.AsOdcmMethod().Overloads);
+#>
+<# bool isPost = IsPostRequestType(c.AsOdcmMethod()); #>
+<#=CreateClassDef(BaseTypeRequestBuilder(c),  RequestBuilderSuperClass(c))#>
 
-<#=CreateClassDef(BaseTypeRequestBuilder(c),  "BaseRequestBuilder")#>
-
-<# foreach (var p in c.AsOdcmMethod().Parameters) { #>
-    public final <#=ParamType(p)#> <#=FieldName(p)#>;
-<# } #>
-
+<# foreach (var method in methods) { #>
     /**
      * The request builder for this <#=TypeName(c)#>
      *
      * @param requestUrl The request url
      * @param client The service client
      * @param options The options for this request
      */
-    public <#=BaseTypeRequestBuilder(c)#>(final String requestUrl, final <#=IBaseClientType()#> client, final List<Option> options<#=MethodParametersSignature(c)#>) {
+    public <#=BaseTypeRequestBuilder(c)#>(final String requestUrl, final <#=IBaseClientType()#> client, final List<Option> options<#=MethodParametersSignature(c.AsOdcmMethod())#>) {
         super(requestUrl, client, options);
-<# foreach (var p in c.AsOdcmMethod().Parameters) { #>
-        <#=FieldName(p)#> = <#=ParamName(p)#>;
+<# foreach (var p in method.Parameters) { #>
+  <# if (isPost) { #>
+      mBodyParams.put("<#=ParamName(p)#>", <#=ParamName(p)#>);
+  <# } else { #>
+      mFunctionOptions.add(new FunctionOption("<#=ParamName(p)#>", <#=ParamName(p)#>));
+  <# } #>
 <# } #>
     }
+<# } #>
```

Comment:
Nit: I liked seeing all of the <# markers on the beginning of the line and indent the C# code to its proper indentation level for readability